.github/workflows/run-cluster.yaml

```diff
@@ -157,6 +162,12 @@ jobs:
       run: |
         source .venv/bin/activate
         ray down .github/assets/ray.yaml -y
+    - name: Upload output dir
+      if: always()
```

Comment:
genuine question, what's the purpose of `if: always()`? isnt that the default behavior? _(i have no idea)_

Reply:
So in a sequence of GitHub Actions steps, if one fails, then *all* subsequent steps by default will be cancelled. Thus, if you annotate a step with `if: always()`, it will always run, even if a previous step returned a non-zero exit code.